terraform/environment/load_balancer_front.tf

```diff
@@ -98,6 +98,46 @@ resource "aws_lb_listener_certificate" "front_loadbalancer_live_service_certific
   certificate_arn = data.aws_acm_certificate.public_facing_certificate.arn
 }
 
+resource "aws_ssm_parameter" "maintenance_switch" {
+  name            = "${local.environment}_enable_maintenance"
+  type            = "String"
+  value           = "false"
+  description     = "values of either 'true' or 'false' only"
+  allowed_pattern = "^(true|false)"
+  overwrite       = true
+  lifecycle {
+    ignore_changes = [value]
+  }
+}
+
+# maintenance site switching
+
+resource "aws_lb_listener_rule" "front_maintenance" {
+  listener_arn = aws_lb_listener.front_loadbalancer.arn
+  priority     = 101 # Specifically set so that maintenance mode scripts can locate the correct rule to modify
+  action {
+    type = "fixed-response"
+
+    fixed_response {
+      content_type = "text/html"
+      message_body = file("${path.module}/maintenance/maintenance.html")
+      status_code  = "503"
+    }
+  }
+
+  condition {
```

Comment:
As I'm not familiar, could you explain how is works at a high level? I can see that you're setting a fixed response on the load balancer, but I'm not sure how setting the maintenance switch tells the site to show the maintenance page. Is it that you're changing the condition on the rule (from the bash script) so that any request with a particular value will trigger the rule that shows the maintenance page?

Reply:
High level is 
- we set up a Load balancer rule that gives a fixed response. 
- in Terraform we ignore condition lifecycle changes, so that we can modify using the script
- the script modifys the rule to enable the fixed response or disables it accordingly, based on the bash script, allowing quick switch over.
I have noticed we might not need the SSM  parameter, i'll test that again without...